### ISSUE TITLE
Side effects function calls in parameters are dropped

### ISSUE THREAD

**Issue by manucorporat** [rollup/rollup#2922]
We have never reproduced this issue on the wild, but I thought it could be a problem after going though the rollup's code. Low priority.

- Rollup Version: 1.14.6
- Operating System (or Browser): browser
- Node Version:

### How Do We Reproduce?

https://rollupjs.org/repl?version=1.14.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiU1Q25mdW5jdGlvbiUyMHNpZGVFZmZlY3RzKCklMjAlN0IlNUNuJTVDdGNvbnNvbGUubG9nKCdwcmludCUyMG1lc3NhZ2UnKSUzQiU1Q24lNUN0cmV0dXJuJTIwdHJ1ZSUzQiU1Q24lN0QlNUNuJTVDbmZ1bmN0aW9uJTIwaG9sYShhJTJDJTIwYiklMjAlN0IlNUNuJTVDdGNvbnNvbGUubG9nKGEpJTNCJTVDbiU3RCU1Q24lNUNuaG9sYSgxJTJDJTIwc2lkZUVmZmVjdHMoKSklM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

### Expected Behavior

It should not drop the call to sideEffects(), since it might have side effects, it's not safe to drop.
Giving this entry-point:
```js 
function sideEffects() {
	console.log('print message');
	return true;
}

function hola(a, b) {
	console.log(a);
}

hola(1, sideEffects());
```

it should emit:

```js
function sideEffects() {
	console.log('print message');
	return true;
}

function hola(a, b) {
	console.log(a);
}

hola(1, sideEffects());
```

### Actual Behavior

It drops the side effect function if the parameter is unused.

```js
function hola(a, b) {
	console.log(a);
}

hola(1);
```

so the console.log() is never printed

**Comment by lukastaegert** [comment]
Interesting, I distinctly remember having written code to prevent this. Will need to check again and add more tests.

**Comment by lukastaegert** [comment]
Thanks again for finding this, the fix was indeed as simple as an `if` instead of an `else if`. And a missed test case. Fix at #2924

**Comment by manucorporat** [comment]
Awesome job @lukastaegert ! you rock
